Server/Repository/PostRepository.cs

```diff
@@ -1,149 +1,127 @@
-﻿using Server.Models;
+﻿using Microsoft.EntityFrameworkCore;
+using PostApp.Models;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
 
-namespace Server.Repository
+namespace PostApp.Repository
 {
     public class PostRepository : IPostRepository
     {
-        public Post Insert(Post post)
+        private readonly fbContext context;
+        public PostRepository(fbContext context)
         {
-            using (fbContext fb = new fbContext())
-            {
-                try
-                {
-                    post.CreatedDate = DateTime.Now;
-                    post.UpdatedDate = null;
-                    post.Likes = 0;
-                    post.Hearts = 0;
-                    fb.Posts.Add(post);
-                    fb.SaveChanges();
-                    return post;
-                }
+            this.context = context;
+        }
 
-                catch (Exception ex)
-                {
-                    throw ex;
-                }
-            }
+        public async Task<Post> Insert(Post post)
+        {
+            post.CreatedDate = DateTime.Now;
+            post.UpdatedDate = null;
+            post.Likes = 0;
+            post.Hearts = 0;
+            context.Posts.Add(post);
+            await context.SaveChangesAsync();
+            return post;
         }
 
-        public List<Post> GetAll()
+        public async Task<List<Post>> GetAll()
         {
-            using (fbContext fb = new fbContext())
-            {
-                return fb.Posts.ToList();
-            }
+            return await context.Posts.ToListAsync();
         }
 
-        public Post GetById(int id)
+        public async Task<Post> GetById(int id)
         {
-            using (fbContext fb = new fbContext())
+            try
             {
-                try
-                {
-
-
-                    return fb.Posts.First(x => x.PostId == id);
-
-                }
-                catch (Exception ex)
-                {
-                    throw ex;
-                }
+                return await context.Posts.FirstAsync(x => x.PostId == id);
```

Comment:
Can we try to use singleordefault and change to other places if yo get only single record

Reply:
changed to SingleOrDefault